src/eval/RealRationalNumber.hpp

```diff
@@ -37,23 +39,24 @@ class RealRationalNumber {
 	template<class T>
 	bool is() const;
 
-	typedef std::function<math::Rational(const math::Rational&, const math::Rational&)> RationalOperation;
-	typedef std::function<math::Real(const math::Real&, const math::Real&)> RealOperation;
+	static RealRationalNumber doPlus(
+			const RealRationalNumber& x,
+			const RealRationalNumber& y);
 
-	static RealRationalNumber doOperation(
+	static RealRationalNumber doTimes(
 			const RealRationalNumber& x,
-			const RealRationalNumber& y,
-			RationalOperation rationalOperation,
-			RealOperation realOperation);
+			const RealRationalNumber& y);
+
 private:
-	typedef boost::variant<math::Real, math::Rational> Number;
+	typedef boost::variant<math::Real, math::Rational, math::ComplexReal, math::ComplexRational> Number;
```

Comment:
Now that this holds Complexes as well, maybe we can come up with some better name instead of RealRationalNumber. Maybe `TypedNumber`?